hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/upgrade/ReconLayoutVersionManager.java

```diff
@@ -47,12 +48,12 @@ public class ReconLayoutVersionManager {
   private int currentMLV;
 
   public ReconLayoutVersionManager(ReconSchemaVersionTableManager schemaVersionTableManager,
```

Comment:
Also in this class while you finalize layout features here in `finalizeLayoutFeatures`, I found some edge case in old code when this layout framework was developed. After calling `execute` method of each action, we are updating the schema version in Recon SQL table. So both are in same try catch block and either of it can fail. If `execute` is success, but updating schema version fails, then also we are throwing the exception with log , but not reverting the action done by `execute`. So this is half baked inconsitency state of Recon. Pls check that part to make it robust.

Reply:
Will be addressed by :- https://issues.apache.org/jira/browse/HDDS-11716